### PR TITLE
Get new navigation in production

### LINKED ISSUE #1355: Update main navigation
Make it...

- more smooth 
- functionally better
- work better on mobile

And add new pages if any.

### PR DESCRIPTION
**Pull request by RonaldDijkstra**
Obviously, the main navigation does a lot for the experience a visitor has. So the last months when we where talking about getting the websites ready for 2019 I thought the navigation would be a great start. 

I was when it was first released proud on how the dropdown and mobile navigation worked, but always thought it could be done a little better.  For example, the animation when switching between dropdowns was a bit off. 

This PR improves everything a freaking lot. 

---
- [x] Remove old javascript
- [x] Add new icons
- [x] Remove old main navs
- [x] Remove old now unneeded assets
- [x] Add new files
- [x] Add new javascript
- [x] Add dropdown
- [x] Add primary
- [x] Add helpers
- [x] Add mobile
- [x] Remove old styling 
- [x] Add new styling
- [x] Add translations

---
- [x] Test in legacy browsers
---

Close #1355

### CODE REVIEW

File: helpers/language_switchers.rb [hunk 1]
```diff
@@ -36,15 +36,38 @@ def page_flag_url(lang)
...
+        locale: lang,
+        class: "link-container with-icon"
+      )
+      html << "</li>"
+    end
+    html
+  end
+
+  # Build the langague_switcher links for desktop navigation
+  def language_switcher_mobile
```
[the remark this answers is not in the document]
Comment (RonaldDijkstra) [reply helpers/language_switchers.rb]: Acceptable

File: helpers/language_switchers.rb [hunk 2]
```diff
@@ -36,15 +36,38 @@ def page_flag_url(lang)
...
 
-  # Build the langague_switcher links
-  def language_switcher
+  # LinkTitle
+  def h3(lang)
+    "<h3 class='link-title link-icon'>#{flag_image(lang)}#{flag_titles[lang]}</h3>"
+  end
+
+  # Build the langague_switcher links for desktop navigation
+  def language_switcher_desktop
```
[the remark this answers is not in the document]
Comment (RonaldDijkstra) [reply helpers/language_switchers.rb]: Acceptable

File: helpers/language_switchers.rb [hunk 3]
```diff
@@ -36,15 +36,38 @@ def page_flag_url(lang)
     end
   end
 
-  # Build the langague_switcher links
-  def language_switcher
+  # LinkTitle
+  def h3(lang)
+    "<h3 class='link-title link-icon'>#{flag_image(lang)}#{flag_titles[lang]}</h3>"
```
[the remark this answers is not in the document]
Comment (RonaldDijkstra) [reply helpers/language_switchers.rb]: Acceptable

**Comment by RonaldDijkstra** [comment]
Deploy preview: https://staging.defacto.nl/

**Comment by RonaldDijkstra** [comment]
The svgs in products should not be inlined

**Comment by RonaldDijkstra** [comment]
Still deployed to: staging.defacto.nl

File: source/localizable/header/_nav-dropdowns.erb [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+
+          <ul class="products-group">
+
+            <% if root_locale == :nl %>
+
+              <li>
+                <% nav_link_to("/capp-lms.html", class: "link-container item-capp") do %>
+                  <%= inline_svg "icons/capp12.svg", title: "CAPP LMS" %>
+                  <div class="product-link-content">
+                    <h3 class="link-title">
```
Comment (sn3p) [inline source/localizable/header/_nav-dropdowns.erb]: Why make these h3 titles?

File: source/localizable/header/_nav-dropdowns.erb [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+    </div>
+
+    <div class="dropdown-section left" data-dropdown="service">
+
+      <div class="dropdown-content">
+
+        <div class="link-group">
+
+          <% nav_link_to("/implementaties.html", class: "link-container with-icon") do %>
+            <h3 class="link-title link-icon">
```
Comment (sn3p) [inline source/localizable/header/_nav-dropdowns.erb]: Why make these h3 titles?

File: helpers/language_switchers.rb (hunk 3 above)
Comment (sn3p) [inline helpers/language_switchers.rb]: idem

**Comment by sn3p** [comment]
screenshot 2018-11-26 at 17 10 50

1. Think its nicer to have the icons aligned to the top (aside the title)
2. Titles are really thin, make them (semi)bold for better readability?

File: source/localizable/header/_nav-mobile.erb [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+              </h4>
+            </li>
+
+            <li>
+              <%= nav_link_to t("header.about"), "/over-ons.html", class: "item-about-us" %>
+            </li>
+
+            <% if root_locale == :nl %>
+              <li>
+                <%= nav_link_to "Service & Support", "/", class: "item-service" %>
```
Comment (sn3p) [inline source/localizable/header/_nav-mobile.erb]: `/support` ?

**Comment by sn3p** [comment]
Should the language switcher work on staging? They all link to `https://staging.defacto.nl`

File: source/stylesheets/pages/_capp-agile-learning.scss [hunk 7]
```diff
@@ -14,11 +14,12 @@ body.capp-agile-learning {
     }
 
     .hero-inner {
-      padding: 2em 1em 4em;
+      // padding: 2em 1em 4em;
```
Comment (sn3p) [inline source/stylesheets/pages/_capp-agile-learning.scss]: Can be removed? ;)

**Comment by RonaldDijkstra** [comment]
> Should the language switcher work on staging? They all link to https://staging.defacto.nl

Something in locale_link_to resolves to that. Should be okay for production.

**Comment by RonaldDijkstra** [comment]
#1620

File: source/localizable/header/_nav-dropdowns.erb (hunk 5 above)
Comment (RonaldDijkstra) [reply source/localizable/header/_nav-dropdowns.erb]: Because they are sort of headings, but I guess this isn't best practice here.

Comment (RonaldDijkstra) [reply source/localizable/header/_nav-dropdowns.erb]: Now using span anywhere.

**Comment by EasterPeanut** [comment]
The heading in the thanks page needs some space:
screenshot 2018-11-27 at 13 04 07

**Comment by RonaldDijkstra** [comment]
Wow thank you

**Comment by EasterPeanut** [comment]
For mobile, a user can now scroll out of the dropdown menu: 
screenshot 2018-11-27 at 14 18 56

File: source/localizable/header/_nav-dropdowns.erb [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+                      <%= t("products.capp.capp-lms") %>
+                    </span>
+                    <p class="link-sub">
+                      <%= t("nav.capp-lms") %>
+                    </p>
+                  </div>
+                <% end %>
+              </li>
+
+              <% else %>
```
Comment (EasterPeanut) [inline source/localizable/header/_nav-dropdowns.erb]: indent

File: source/localizable/header/_nav-mobile.erb [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+              <% end %>
+
+            </ul>
+
+          </div>
+
+        </div>
+
+        <div class="mobile-demo">
+          <%= link_to t("actions.request_demo"), "/demo-aanvragen.html", class: " button rounded pink item-demo" %>
```
Comment (EasterPeanut) [inline source/localizable/header/_nav-mobile.erb]: a space too many `" button`?

**Comment by RonaldDijkstra** [comment]
> Is that meant to happen?

At this point yes. In the navigation as now in production it partly breaks the standard scroll-to-top function which I do not like. I made an issue to make this better and stick.

File: source/javascripts/_navigation.js [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+    return t || (t = document.body), Array.prototype.slice.call(t.querySelectorAll(event));
+  },
+
+  ready: function (event) {
+    document.readyState !== 'loading' ? event() : document.addEventListener('DOMContentLoaded', event);
+  },
+};
+
+Stut.isRetina = window.devicePixelRatio > 1.3,
+Stut.mobileViewportWidth = 760,
```
Comment (EasterPeanut) [inline source/javascripts/_navigation.js]: Not sure but it seems that `isRetina` and `isMobileViewport` arent used anywhere

File: source/stylesheets/layout/_header.scss [hunk 11]
```diff
@@ -6,197 +6,828 @@
...
 
-      &.header {
-        .brand-mark {
-          fill: $orange-logo;
-        }
+  // Styling for aria current = page
+  // https://github.com/thoughtbot/middleman-aria_current
+  a[aria-current]:not([aria-current="false"]) {
+    font-weight: 700;
+
```
[the remark this answers is not in the document]
Comment (RonaldDijkstra) [reply source/stylesheets/layout/_header.scss]: I like my trailing whitespaces.

**Comment by RonaldDijkstra** [comment]
I've done all things you reviewed and some more ;-) deployed again to https://staging.defacto.nl

**Comment by RonaldDijkstra** [comment]
```ruby
def full_url(url, locale = I18n.locale)
    base =
      if staging
        "https://staging.defacto.nl"
      else
```

Here is what goes wrong with the language switcher on staging.

**Comment by RonaldDijkstra** [comment]
Close #1620

**Comment by RonaldDijkstra** [comment]
Ready for 🚢

File: helpers/links.rb [hunk 12]
```diff
@@ -19,11 +19,16 @@ def full_blog_url(lang)
...
 
-  # Get full url
-  def full_url(url, locale = I18n.locale)
+  # Get full cases url
+  def full_cases_url(lang)
+    lang == :en ? "/" : full_url("/cases/", lang)
+  end
+
+  # Get full current url
+  def full_env_url(url, locale = I18n.locale)
```
Comment (sn3p) [inline helpers/links.rb]: Is `full_env_url` only used for `<meta property="og:*">` tags?

Comment (RonaldDijkstra) [reply helpers/links.rb]: Yes

Comment (sn3p) [reply helpers/links.rb]: Do we really need this just for staging?

Comment (RonaldDijkstra) [reply helpers/links.rb]: But it could be in the full_url method with some options. What I did here is just a quick fix.

**Comment by sn3p** [comment]
Alignment is a bit off here, looks like the icon is larger and the text is pushed to the right:

screenshot 2018-11-30 at 14 04 08

For another PR, but some descriptions and with a `.` and some not:

screenshot 2018-11-30 at 14 03 30

**Comment by RonaldDijkstra** [comment]
Thanks, addressed all things you said and deployed again to staging

**Pull request merged**